pkg/trace/stats/concentrator.go

```diff
@@ -187,16 +195,34 @@ func (c *Concentrator) flushNow(now int64) pb.StatsPayload {
 	c.mu.Unlock()
 	sb := make([]pb.ClientStatsPayload, 0, len(m))
 	for k, s := range m {
-		sb = append(sb, pb.ClientStatsPayload{
-			Env:      k.Env,
-			Hostname: k.Hostname,
-			Version:  k.Version,
-			Stats:    s,
-		})
+		p := pb.ClientStatsPayload{
+			Env:         k.Env,
+			Hostname:    k.Hostname,
+			ContainerID: k.ContainerID,
+			Version:     k.Version,
+			Stats:       s,
+		}
+		resolveContainerTags(&p)
+		sb = append(sb, p)
 	}
 	return pb.StatsPayload{Stats: sb, AgentHostname: c.agentHostname, AgentEnv: c.agentEnv, AgentVersion: info.Version}
 }
 
+// resolveContainerTags takes any ContainerID found in p to fill in the appropriate tags.
+func resolveContainerTags(p *pb.ClientStatsPayload) {
+	ctags, err := tagger.Tag("container_id://"+p.ContainerID, collectors.HighCardinality)
```

Comment:
@markadev are we sticking to high cardinality here or should we go lower? See https://github.com/DataDog/datadog-agent/blob/048f714e0477ab5005d14ad6888cdbb82fd17384/pkg/tagger/collectors/types.go#L49-L60

Reply:
If we only fetch the `High` tags then we wouldn't get the real container ID in the `container_id` tag.  https://github.com/DataDog/datadog-agent/blob/c002cc45f266d639e1d8ad4d9923c782b5107452/pkg/tagger/collectors/ecs_fargate_extract.go#L75

Reply:
Good point. Thanks. Resolving.